sklearn/utils/estimator_checks.py

```diff
@@ -2220,10 +2220,9 @@ def check_classifiers_predictions(X, y, name, classifier_orig):
                                 ", ".join(map(str, y_pred))))
 
     # training set performance
-    if name != "ComplementNB":
-        # This is a pathological data set for ComplementNB.
-        # For some specific cases 'ComplementNB' predicts less classes
-        # than expected
+    if not _safe_tags(classifier, key="poor_score"):
+        # In some cases poor_score estimators are predicting
+        # less classes than expected
         assert_array_equal(np.unique(y), np.unique(y_pred))
```

Comment:
Currently, our definition of "poor_score" only for the test set.

> whether the estimator fails to provide a "reasonable" test-set score

In this case, we are checking the predictions on training set. I would be okay with extending the definition of `poor_score` to cover the training set.